code/modules/mob/living/simple_animal/revenant/revenant_abilities.dm

```diff
@@ -0,0 +1,318 @@
+//Harvest Essence: The bread and butter of the revenant. The basic way of harvesting additional essence.
+/obj/effect/proc_holder/spell/targeted/revenant_harvest
+	name = "Harvest (0E)"
+	desc = "Siphons the lingering spectral essence from a human, empowering yourself."
+	panel = "Revenant Abilities"
+	charge_max = 100 //Short cooldown
+	clothes_req = 0
+	range = 5
+	var/essence_drained = 0
+	var/draining
+	var/list/drained_mobs = list() //Cannot harvest the same mob twice
+
+/obj/effect/proc_holder/spell/targeted/revenant_harvest/cast(list/targets, var/mob/living/simple_animal/revenant/user = usr)
+	if(user.inhibited)
+		user << "<span class='warning'>Something is blocking the use of [src]!</span>"
+		charge_counter = charge_max
+		return
+	for(var/mob/living/carbon/human/target in targets)
+		spawn(0)
+			if(draining)
+				user << "<span class='warning'>You are already siphoning the essence of a soul!</span>"
+				return
+			if(target in drained_mobs)
+				user << "<span class='warning'>[target]'s soul is dead and empty.</span>"
```

Comment:
Yes, it does. It is for balance reasons that every revenant can only harvest one time so their essence doesn't skyrocket, but it's also intended for new revenants to be able to get essence from other drained corpses too.